### PR TITLE
Add buttons to Notebooks for creating new cells between other existing cells

### PR DESCRIPTION
**Pull request by corivera**
image

image

### CODE REVIEW

File: src/sql/workbench/parts/notebook/browser/cellToggleMoreActions.ts [hunk 1]
```diff
@@ -30,10 +30,6 @@ export class CellToggleMoreActions {
 		@IInstantiationService private instantiationService: IInstantiationService) {
 		this._actions.push(
 			instantiationService.createInstance(DeleteCellAction, 'delete', localize('delete', "Delete")),
-			instantiationService.createInstance(AddCellFromContextAction, 'codeBefore', localize('codeBefore', "Insert Code Before"), CellTypes.Code, false),
```
Comment (kevcunnane) [inline src/sql/workbench/parts/notebook/browser/cellToggleMoreActions.ts]: Can we create a task to track need for keyboard shortcuts to do these actions? Moving from right-click to new inline set of actions, makes it harder for accessibility. We can either update Add Cell command or add new accelerators for all these.

Again: not needed in this PR, but we should track & prioritize doing this.

File: src/sql/workbench/parts/notebook/browser/notebook.component.html [hunk 2]
```diff
@@ -9,11 +9,35 @@
...
-		<div class="notebook-cell" *ngFor="let cell of cells" (click)="selectCell(cell, $event)" [class.active]="cell.active">
-			<code-cell-component *ngIf="cell.cellType === 'code'" [cellModel]="cell" [model]="model" [activeCellId]="activeCellId">
-			</code-cell-component>
-			<text-cell-component *ngIf="cell.cellType === 'markdown'" [cellModel]="cell" [model]="model" [activeCellId]="activeCellId">
-			</text-cell-component>
+		<div class="hoverButtonsContainer" *ngIf="(cells && cells.length > 0) && !isLoading">
+			<span class="containerBackground"></span>
+			<button class="hoverButton" (click)="addCell('code', 0)">
+				<div class="addCodeIcon"></div>
+				<span>Add code</span>
```
Comment (kevcunnane) [inline src/sql/workbench/parts/notebook/browser/notebook.component.html]: These strings need to be localized. Please provide the string as a getter in the code behind and use `{{addCodeMsg}}` or similar to include them

File: src/sql/workbench/parts/notebook/browser/notebook.css [hunk 3]
```diff
@@ -136,4 +136,76 @@
...
+	background-position: center center;
+	background-repeat: no-repeat;
+	width: 10px;
+	height: 10px;
+	background-size: contain;
+	margin-right: 4px;
+}
+
+.notebookEditor .hoverButton .addCodeIcon {
+	background-image: url("./media/light/add_code.svg");
```
Comment (kevcunnane) [inline src/sql/workbench/parts/notebook/browser/notebook.css]: please check with Anthony on relative links, he's been making a lot of changes around this in his code layering.

File: src/sql/workbench/parts/notebook/browser/notebook.css [hunk 4]
```diff
@@ -136,4 +136,76 @@
...
+.notebookEditor .hoverButtonsContainer {
+	display: flex;
+	flex-direction: row;
+	align-items: center;
+	justify-content: center;
+	width: 40%;
+	margin: 1px auto;
+}
+
+.notebookEditor .hoverButtonsContainer .containerBackground {
```
Comment (kevcunnane) [inline src/sql/workbench/parts/notebook/browser/notebook.css]: I think this should also be invisible until we hover.

Comment (corivera) [reply src/sql/workbench/parts/notebook/browser/notebook.css]: Fixed

File: src/sql/workbench/parts/notebook/browser/notebook.component.html (hunk 2 above)
Comment (corivera) [reply src/sql/workbench/parts/notebook/browser/notebook.component.html]: Fixed

File: src/sql/workbench/parts/notebook/browser/notebook.css (hunk 3 above)
Comment (corivera) [reply src/sql/workbench/parts/notebook/browser/notebook.css]: @anthonydresser Any issues we might need to worry about here?

File: src/sql/workbench/parts/notebook/browser/cellToggleMoreActions.ts (hunk 1 above)
Comment (corivera) [reply src/sql/workbench/parts/notebook/browser/cellToggleMoreActions.ts]: Doesn't look like we had these in the right click menu, just the overflow menu in the corner of the cell. We have existing shortcuts for adding new cells: Ctrl+Shift+C for code, Ctrl+Shift+T for text.

File: src/sql/workbench/parts/notebook/browser/notebook.css (hunk 3 above)
Comment (anthonydresser) [reply src/sql/workbench/parts/notebook/browser/notebook.css]: I can't think of any.

**Pull request merged**